### PR TITLE
[WIP] Support Go Modules and update vendor directory

### PR DESCRIPTION
**Pull request by terrytangyuan**
This supports Go Modules as mentioned in the ROADMAP. The diff is large but most of them are changes in the auto-generated vendor folder. When reviewing, focus on changes in `.travis.yaml`, `go.mod`, and `go.sum`.

Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/169)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ language: go
 go_import_path: github.com/kubeflow/mpi-operator
 
 go:
-- 1.11.4
+- 1.13
```
Comment (rongou) [inline .travis.yml]: Also update the `Dockerfile` to use 1.13?

Comment (terrytangyuan) [reply .travis.yml]: Updated. Although I am still trying to figure out why Travis is failing.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.12.7-alpine3.10 AS build
+FROM golang:1.13.0-alpine3.10 AS build
```
Comment (rongou) [inline Dockerfile]: The latest version is `1.13.6-alpine3.11`.

Comment (rongou) [reply Dockerfile]: Or maybe we just use `alpine` which always points to the latest version.

Comment (terrytangyuan) [reply Dockerfile]: Thanks. Updated to `1.13.6-alpine3.11`. I think more explicit is better.

**Comment by terrytangyuan** [comment]
I am marking this as WIP for now until I find fix for the CI failure.

**Comment by terrytangyuan** [comment]
@k82cn I am trying to figure out a solution for the following failure related to kube-batch. Any suggestions?
```
FAIL	github.com/kubeflow/mpi-operator/pkg/controllers/v1alpha2 [build failed]
FAIL
exit status 2: warning: no packages being tested depend on matches for pattern github.com/kubeflow/mpi-operator/pkg/controllers/v1alpha1
warning: no packages being tested depend on matches for pattern github.com/kubeflow/mpi-operator/pkg/version
# github.com/kubernetes-sigs/kube-batch/pkg/client/clientset/versioned/typed/scheduling/v1alpha1/fake
../../../../pkg/mod/github.com/kubernetes-sigs/kube-batch@v0.4.2/pkg/client/clientset/versioned/typed/scheduling/v1alpha1/fake/fake_podgroup.go:134:44: not enough arguments in call to testing.NewPatchSubresourceAction
	have (schema.GroupVersionResource, string, string, []byte, []string...)
	want (schema.GroupVersionResource, string, string, types.PatchType, []byte, ...string)
../../../../pkg/mod/github.com/kubernetes-sigs/kube-batch@v0.4.2/pkg/client/clientset/versioned/typed/scheduling/v1alpha1/fake/fake_queue.go:115:48: not enough arguments in call to testing.NewRootPatchSubresourceAction
	have (schema.GroupVersionResource, string, []byte, []string...)
	want (schema.GroupVersionResource, string, types.PatchType, []byte, ...string)
```

**Comment by k82cn** [comment]
no clue for now :( , did you try [Volcano](http://github.com/volcano-sh/volcano) ?

**Comment by terrytangyuan** [comment]
@k82cn I'll catch up with Volcano later.

The kube-batch issue seems to have been fixed via https://github.com/kubernetes-sigs/kube-batch/commit/a0a18bb7eccdce1c44ba13dab77c0a54dd1329f1#diff-42f47f5106e321f47d598ae905aad87d. However, Travis CI still seems to be using the old kube-batch version (perhaps wrongly cached). I'll go ahead and merge this to see if this will be fixed once it's on master.

**Comment by k82cn** [comment]
> @k82cn I'll catch up with Volcano later.

Great! Thanks :)

**Pull request merged**